### PR TITLE
projection_subtraction_protocol: adapt for the new implementation

### PR DESCRIPTION
**Pull request by EstrellaFG**
Adapt projection subtraction and its scipion tests for the new implementation. This PR should not be merged before https://github.com/I2PC/xmipp/pull/563

### CODE REVIEW

File: xmipp3/tests/test_protocols_xmipp_3d.py [hunk 1]
```diff
@@ -1476,12 +1476,6 @@ def testXmippProjSub(self):
         self.launchProtocol(protCreateGallery)
```
Comment (albertmena) [inline xmipp3/tests/test_protocols_xmipp_3d.py:1476]: Could you reorder it adding a loop or adding some comments to divide each test, is difficult to follow it and maintenance it.

Comment (albertmena) [inline xmipp3/tests/test_protocols_xmipp_3d.py:1476]: Could you reorder it adding a loop or adding some comments to divide each test, is difficult to follow it and maintenance it.

File: xmipp3/protocols/protocol_subtract_projection.py [hunk 2]
```diff
@@ -29,15 +29,15 @@
...
 from pwem.protocols import EMProtocol
-from xmipp3.convert import writeSetOfParticles
+from xmipp3.convert import writeSetOfParticles, readSetOfParticles
 
 
 class XmippProtSubtractProjection(EMProtocol):
-    """ This protocol computes the subtraction between particles and a initial volume, by computing its projections
-    with the same angles that input particles have. Then, each particle and the correspondent projection of the initial
-    volume are numerically adjusted and subtracted using a mask which denotes the region to keep. """
+    """ This protocol computes the subtraction between particles and a reference volume, by computing its projections
```
Comment (albertmena) [inline xmipp3/protocols/protocol_subtract_projection.py:38]: The line break makes difficult to read in the window help

**Review by albertmena — CHANGES_REQUESTED** [review]

File: xmipp3/protocols/protocol_subtract_projection.py [hunk 3]
```diff
@@ -46,27 +46,22 @@ class XmippProtSubtractProjection(EMProtocol):
...
+        form.addParam('limit_freq', BooleanParam, label="Limit frequency?: ", default=False, expertLevel=LEVEL_ADVANCED,
+                      help='Limit frequency in the adjustment process to the frequency correspondent to the resolution'
+                           ' indicated in "Maximum resolution" field above')
+        form.addParam('sigma', FloatParam, label="Decay of the filter (sigma): ", default=3, expertLevel=LEVEL_ADVANCED,
+                      help='Decay of the filter (sigma) to smooth the mask transition')
+        form.addParam('pad', IntParam, label="Fourier padding factor: ", default=2, expertLevel=LEVEL_ADVANCED,
+                      help='The volume is zero padded by this factor to produce projections')
 
     # --------------------------- INSERT steps functions --------------------------------------------
     def _insertAllSteps(self):
```
[the remark this answers is not in the document]
Comment (EstrellaFG) [reply xmipp3/protocols/protocol_subtract_projection.py:65]: I think the protocol has this structure already: 
1. convertStep (equivalent to readInput)
2. subtractionStep (RunCosas -> It is a very short method, I do not think I could divide it more)
3. createOutputStep

[the remark this answers is not in the document]
Comment (albertmena) [reply xmipp3/protocols/protocol_subtract_projection.py:65]: Yes, sorry.

**Review by albertmena — APPROVED** [review]

**Review by albertmena — APPROVED** [review]

**Review by albertmena — CHANGES_REQUESTED** [review]

**Review by albertmena — APPROVED** [review]

**Pull request merged**